File: LeetCode/299_bulls_and_cows.rb
# @param {String} secret
# @param {String} guess
# @return {String}
def get_hint(secret, guess)
    possible_cows = Hash.new(0)
    
    total_cows = 0
    total_bulls = 0
    
    secret.each_char.each_with_index do |secret_character, index| 
        if guess[index] == secret_character
            total_bulls += 1
        else
            possible_cows[secret_character] += 1
        end
    end
    
    guess.each_char.each_with_index do |guess_character, index|
        if possible_cows[guess_character] != 0 and guess_character != secret[index]
            total_cows += 1
            possible_cows[guess_character] -= 1
        end
    end
    
    return "#{total_bulls}A#{total_cows}B"
end

File: LeetCode/299_bulls_and_cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        int possibleCows[10] = {0,0,0,0,0,0,0,0,0,0};
        int cows = 0;
        int bulls = 0;
        
        for(size_t index = 0; index < secret.length(); ++index){
            if(secret[index] == guess[index]){
                ++bulls;
            } else {
                ++possibleCows[secret[index] - '0'];
            }
        }
        
        for(size_t index = 0; index < secret.length(); ++index){
            if(secret[index] != guess[index] and possibleCows[guess[index] - '0'] > 0){
                ++cows;
                --possibleCows[guess[index] - '0'];
            }
        }
        
        return to_string(bulls) + "A" + to_string(cows) + "B";
        
    }
};